recipe/install-conda-specs.sh

```diff
@@ -0,0 +1,31 @@
+set -ex
+export CHOST="${gcc_machine}-${gcc_vendor}-linux-gnu"
+specdir=$PREFIX/lib/gcc/$CHOST/${gcc_version}
+if [[ "$cross_target_platform" == "$target_platform" ]]; then
+    install -Dm644 -T ${SRC_DIR}/build/gcc/specs $specdir/conda.specs
+
+    # Add specs when we're not cross compiling so that the toolchain works more like a system
+    # toolchain (i.e. conda installed libs can be #include <>'d and linked without adding any
+    # cmdline args or FLAGS and likewise the assumptions we have about rpath are built in)
+    #
+    # THIS IS INTENDED as a safety net for casual users who just want the native toolchain to work.
+    # It is not to be relied on by conda-forge package recipes and best practice is still to set the
+    # appropriate FLAGS vars (either via compiler activation scripts or explicitly in the recipe)
+    #
+    # We use double quotes here because we want $PREFIX and $CHOST to be expanded at build time
+    #   and recorded in the specs file.  It will undergo a prefix replacement when our compiler
+    #   package is installed.
+    sed -i -e "/\*link_command:/,+1 s+%.*+& %{\!static:-rpath ${PREFIX}/lib -rpath-link ${PREFIX}/lib -disable-new-dtags} -L ${PREFIX}/lib+" $specdir/conda.specs
+    # use -idirafter to put the conda "system" includes where /usr/local/include would typically go
+    # in a system-packaged non-cross compiler
+    sed -i -e "/\*cpp_options:/,+1 s+%.*+& -idirafter ${PREFIX}/include+" $specdir/conda.specs
```

Comment:
> Can you explain why -isystem doesn't work in a comment?

@isuruf, I thought I explained why I used `-dirafter` in a comment already.  `-isystem` might work but it puts `$PREFIX/include` at the beginning of the search path and that isn't where GCC puts the system include paths when it builds in 'non-cross' mode.  If you know that specific things are going to break by using `-idirafter` instead of `-isystem`, I'll change it.  I just feel like `-idirafter` is the "right" place to put `$PREFIX/include` if we're treating it like our own `/usr/local/include` and that is what I'm trying to do.

Reply:
No, on centos7 and I'm pretty sure most system build gcc, `/usr/local/include` and `/usr/include` are after the gcc directories:

```
-bash-4.2$ cat /etc/redhat-release 
CentOS Linux release 7.9.2009 (Core)
-bash-4.2$ echo | /usr/bin/gcc -E -Wp,-v -x c -
ignoring nonexistent directory "/usr/lib/gcc/x86_64-redhat-linux/4.8.5/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../../../x86_64-redhat-linux/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include
 /usr/local/include
 /usr/include
End of search list.
# 1 "<stdin>"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "/usr/include/stdc-predef.h" 1 3 4
# 1 "<command-line>" 2
# 1 "<stdin>"
```

That's why I used `-idirafter`.  And, I've confirmed that even though it is at the end of `*cpp_options:` and `*cc1_options:`, passing `-isystem $PREFIX/include` will move it earlier.  I suppose I should add that as a test.

Reply:
@isuruf I made the changes you asked for and also added tests to make sure that `-isystem $PREFIX/include` added on the cmdline moves `$PREFIX/include` to the start of the search path so that someone can intentionally override gcc libs if they need to.  I still think `-idirafter` is the correct default location for `$PREFIX/include`

Reply:
@isuruf, I think I've got this straightened out with your feedback now.